hamocc/mo_read_pi_ph.F90

```diff
@@ -0,0 +1,189 @@
+    ! Only read PI pH climatology if needed for DMS
+    if(with_dmsph) then
+
+       ! allocate pi_ph_clim field
+       if(.not. allocated(pi_ph_clim)) call alloc_pi_ph_clim(kpie,kpje)
+
+       ! Open netCDF data file
+       IF(mnproc==1) THEN
+          ncstat = NF90_OPEN(trim(pi_ph_file), NF90_NOWRITE, ncid)
+          write(io_stdo_bgc,*) 'HAMOCC: opening PI_PH climatology file'
+          IF (ncstat.NE.NF90_NOERR ) THEN
+             CALL xchalt('(ini_pi_ph: Problem with netCDF1)')
+                    stop '(ini_pi_ph: Problem with netCDF1)'
+          END IF
+       END IF
+       !
+       ! Read  data
+       call read_netcdf_var(ncid,'pH',pi_ph_in(1,1,1),pi_ph_record,0,0)
+       !
+       ! Close file
+       IF(mnproc==1) THEN
+          ncstat = NF90_CLOSE(ncid)
+          IF ( ncstat .NE. NF90_NOERR ) THEN
+             CALL xchalt('(ini_pi_ph: Problem with netCDF200)')
+                    stop '(ini_pi_ph: Problem with netCDF200)'
+          END IF
+       END IF
+
+       ! set missings over land
+       do l=1,pi_ph_record
+          do j=1,kpje
+             do i=1,kpie
+                if(omask(i,j).gt.0.5) then
+                   pi_ph_clim(i,j,l) = pi_ph_in(i,j,l)
+                else
+                   pi_ph_clim(i,j,l) = 0.
+                endif
+             enddo
+          enddo
+       enddo
+    endif
+
+  end subroutine ini_pi_ph
+
+
+  !**********************************************************************
+  ! PUBLIC SUBROUTINE : GET_PI_PH
+  !
+  ! Return PI_PH field for a given month.
+  !**********************************************************************
+  subroutine get_pi_ph(kpie,kpje,kplmon)
+    use mo_control_bgc, only: with_dmsph
+
+    implicit none
+
+    integer, intent(in) :: kpie,kpje,kplmon
+    integer, save :: oldmonth=0
+
+    ! Ensure that pi_ph is allocated
+    if(.not. allocated(pi_ph)) call alloc_pi_ph(kpie,kpje)
+
+    ! Update only if PI pH climatology is used for DMS
+    if(with_dmsph) then
+       if(kplmon /= oldmonth) then
+          pi_ph = reshape(pi_ph_clim(:,:,kplmon), [kpie,kpje])
+          oldmonth = kplmon
+       endif
+    endif
+
+  end subroutine get_pi_ph
+
+
+  !**********************************************************************
+  ! PRIVATE SUBROUTINE : ALLOC_PI_PH
+  !
+  ! Allocate the PI_PH field.
+  !**********************************************************************
+  subroutine alloc_pi_ph(kpie,kpje)
+    use mod_xc,         only: mnproc
+    use mo_control_bgc, only: io_stdo_bgc
+
+    implicit none
+    integer, intent(in) :: kpie,kpje
+    integer             :: errstat
+
+    IF (mnproc.eq.1) THEN
+       WRITE(io_stdo_bgc,*)'Memory allocation for variable pi_ph ...'
+       WRITE(io_stdo_bgc,*)'First dimension    : ',kpie
+       WRITE(io_stdo_bgc,*)'Second dimension   : ',kpje
+    ENDIF
+
+    ALLOCATE (pi_ph(kpie,kpje),stat=errstat)
+    if(errstat.ne.0) stop 'not enough memory pi_ph'
+    pi_ph(:,:) = 0.0
+
+  end subroutine alloc_pi_ph
```

Comment:
I find this unnecessarily complicated, the other fields (dust, ndep) are just local variables in hamocc_step, they are created/removed automatically when hamocc_step is called/finalized.

Reply:
Think I will leave it like this for now. Jerry mentioned that he might introduce a 3D PI_pH at some point in the future. If so, I guess we might need to think again about this structure.